lib/astarte/client/appengine/devices.ex

```diff
@@ -19,6 +19,19 @@
 defmodule Astarte.Client.AppEngine.Devices do
   alias Astarte.Client.{APIError, AppEngine}
 
+  def list(%AppEngine{} = client, query_opts \\ []) do
+    request_path = "devices"
+    tesla_client = client.http_client
+
+    with {:ok, %Tesla.Env{} = result} <- Tesla.get(tesla_client, request_path, query: query_opts) do
+      if result.status == 200 do
+        {:ok, result.body}
```

Comment:
so basically this is not the whole list of devices, but just a chunk of them. there will be anyway a next and a prev pointer, right?

Reply:
also: as soon as we get the first chunk, how do we get the next one? I can't understand how to provide it.

Reply:
Changed `list/2` to return whole list of devices in format {:ok, {"data" => []}} or an error